src/components/sidebar/SafeList/index.tsx

```diff
@@ -75,27 +79,59 @@ const SafeList = ({ closeDrawer }: { closeDrawer?: () => void }): ReactElement =
   const hasNoSafes = Object.keys(ownedSafes).length === 0 && Object.keys(addedSafes).length === 0
   const isWelcomePage = router.pathname === AppRoutes.welcome
 
+  const handleFetchQueued = async () => {
+    const fetchedQueuedTxs: Record<string, string | undefined> = {}
+
+    for (let [chainId, safes] of Object.entries(addedSafes)) {
+      for (let safeAddress of Object.keys(safes)) {
```

Comment:
You could omit the current safe and use `useTxQueue` instead. It's already loaded for the current safe.

Reply:
I've skipped the request for the currentSafe and use the data from `useTxQueue`